src/core/reference/include/ngraph/runtime/reference/scatter_nd_update.hpp

```diff
@@ -39,7 +39,7 @@ void scatterNdUpdate(const dataType* const inputData,
     for (size_t i = 0; i != num_of_updates; ++i) {
         const auto indices_coord = indices + i * indicesShape.back();
         const auto coord = span(indices_coord, indicesShape.back());
-        const auto out_index = std::inner_product(begin(coord), end(coord), begin(input_data_dim_pading), uint64_t(0));
+        const auto out_index = std::inner_product(begin(coord), end(coord), begin(input_data_dim_pading), uint64_t(0)) - 1;
```

Comment:
To exclude memory corruption on different corner cases - it would be great to put check on 0 before extracting 1
and also comment why we need that operation would be beneficial